src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIStructure.java

```diff
@@ -130,6 +130,11 @@
      */
     private int rotation = 0;
 
+    /**
+     * all the fluids to be removed later.
+     */
+    private Map<Integer, List<BlockPos>> fluidsToRemove = new TreeMap<>();
```

Comment:
this needs to be stored in the building class and serialized/deserialized so that it works on restart midway

Reply:
OK i will do that.

Reply:
uhm, in what building class?

Reply:
AbstractBuildingStructureBuilder

Reply:
ok thanks

Reply:
I have tried implementing this, but as i can due to the game crashing when i try to load a world with a working builder(in the dev env only) i can't test it, so i just hope it works.

Reply:
game crashing on startup of the world?

Reply:
In dev mode it should crash less not more =D

Reply:
yes, when i load a world in the dev env and there is a builder in it the game crashes.
i did post that on discord some time in the past(with log), i just didn't make an issue for it because its only in the dev env.

Reply:
can you send this to me (via pm)?

Reply:
yes.

Reply:
This should work now...